### PR TITLE
FMD-156: Allow list action

### PR DESCRIPTION
**Pull request by gidsg**
### Change description
We want to allow list action in order to return an appropriate error if key not found.

- [ ] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)
- [x] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
Will test after deployment that NoSuchKey exception is thrown

### CODE REVIEW

**Review by cyrusdobbs — APPROVED** [review]
LGTM

File: tests/integration_tests/test_retrieve_submission_file.py [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
+def test_buckets():
+    """Sets up and tears down buckets used by this module.
+    On set up:
+    - creates data-store-successful-files-unit-tests
+    On tear down, deletes all objects stored in the buckets and then the buckets themselves.
+    """
+
+    create_bucket(Config.AWS_S3_BUCKET_SUCCESSFUL_FILES)
+    yield
+    delete_bucket(Config.AWS_S3_BUCKET_SUCCESSFUL_FILES)
```
Comment (cyrusdobbs) [inline tests/integration_tests/test_retrieve_submission_file.py:23]: best to use `test_buckets` defined in `conftest.py` (this was added in a recent commit by @MarcUsher)

**Review by cyrusdobbs — CHANGES_REQUESTED** [review]

File: tests/integration_tests/test_retrieve_submission_file.py (hunk 1 above)
Comment (gidsg) [reply tests/integration_tests/test_retrieve_submission_file.py]: Good spot @cyrusdobbs looks like this fixture was no longer needed, so I've removed and force-pushed.

**Review by cyrusdobbs — APPROVED** [review]
LGTM

**Pull request merged**